database/database.go

```diff
@@ -1,127 +1,585 @@
-// Create docker instance with postgres databasemanager inside
-// sudo docker run --name some-postgres -e POSTGRES_PASSWORD=1234 -e POSTGRES_DB=quote_gallery -p 0.0.0.0:5432:5432 -d postgres
-
 package database
 
 import (
 	"database/sql"
+	"errors"
 	"log"
 
 	// loading postgresql driver
 	_ "github.com/lib/pq"
 )
 
-// Teacher struct
-type Teacher struct {
-	ID    int
-	Name  string
-	Title string
-	Note  string
+/* -------------------------------------------------------------------------- */
+/*                                 DEFINITIONS                                */
+/* -------------------------------------------------------------------------- */
+
+// QuoteT stores one quote
+// QuoteID   the unique identificator of the quote
+// TeacherID the unique identifier of the corresponding teacher
+// Context	  the context of the quote
+// text       the text of the quote itself
+// unixtime   optional
+//
+// match  	  only locally, not safed in PostgreSQL database!
+//			  used by GetQuotesFromString to quantify how well this quote fits the string
+type QuoteT struct {
+	QuoteID   uint32
+	TeacherID uint32
+	Context   string
+	Text      string
+	Unixtime  uint64
+	Upvotes   uint32
+	Match     float32
 }
 
-// Quote struct
-type Quote struct {
-	ID      int
-	Teacher Teacher
-	Text    string
+// UnverifiedQuoteT stores one unverified quote
+// QuoteID    the unique identificator of the unverified quote
+// teacher 	   the name of the teacher
+// Context	   the context of the quote
+// text        the text of the quote itself
+// unixtime    optional
+// iphash	   optional
+//
+type UnverifiedQuoteT struct {
+	QuoteID  uint32
+	Teacher  string
```

Comment:
suggestion: change `Teacher` to `TeacherName` to avoid confusion with `Teacher` type
don't we need `TeacherID` as well? either can be given.